src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -864,14 +865,12 @@ public static unsafe int IndexOfAny(ref char searchStart, char value0, char valu
                         // First time this checks again against 0, however we will move into final compare if it fails.
                         while (lengthToExamine > offset)
                         {
-                            search = LoadVector256(ref searchStart, offset);
+                            search = Vector256.LoadUnsafe(ref ushortSearchStart, offset);
                             // Bitwise Or to combine the flagged matches for the second value to our match flags
-                            matches = Avx2.MoveMask(
-                                            Avx2.Or(
-                                                Avx2.CompareEqual(values0, search),
-                                                Avx2.CompareEqual(values1, search))
-                                            .AsByte());
-                            // Note that MoveMask has converted the equal vector elements into a set of bit flags,
+                            matches = (Vector256.Equals(values0, search) | Vector256.Equals(values1, search))
+                                .AsByte().ExtractMostSignificantBits();
+
+                            // Note that ExtractMostSignificantBits has converted the equal vector elements into a set of bit flags,
                             // So the bit position in 'matches' corresponds to the element offset.
                             if (matches == 0)
```

Comment:
For my edification, does using `.AsByte().ExtractMostSignificantBits()` and the doing `== 0` produce more efficient code than using ` == Vector256<ushort>.Zero`?

Reply:
> We do the latter on the Vector128 code path, and I'm wondering why this one differs.

`ExtractMostSignificantBits()` is cheap on x64 but expensive for arm64. `Vector128` includes arm64, so we delay this operation until it's needed for `Vector128`. We don't need to do this for `Vector256`, which currently is supported only by x64.

I've verified that with some experiment in other PR (Tanner asked for that as well)

Reply:
Thanks.  And for the first part of the question:

> does using .AsByte().ExtractMostSignificantBits() and the doing == 0 produce more efficient code than using == Vector256<ushort>.Zero?

Should I take it from your answer that the answer is "yes"?